src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -573,20 +321,176 @@ else if(work.getBaseCaseObj().isUseRegexInstead())
+		    manager.foundDir(url, code, responce, work.getBaseCaseObj().getBaseCase(), rawResponce, work.getBaseCaseObj());
+		}
+		else
+		{
+		    //found a file
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found File (base case)" + url.toString());
+		    }
+		    manager.foundFile(url, code, responce, work.getBaseCaseObj().getBaseCase(), rawResponce, work.getBaseCaseObj());
+		}
+               //manager.foundError(url, "Base Case Mode Error - Responce code came back as " + code + " it should have been 200");
+               //manager.workDone();
+	}
+
+	private void verifyResponseForValidRequests(int code, String responce, String rawResponce) {
+		if(Config.debug)
+		{
+		    System.out.println("DEBUG Worker[" + threadId + "]: Base Case Check " + url.toString());
+		}
+
+
+		//TODO move this option to the Adv options
+		//if the response does not match the base case
+		Pattern regexFindFile = Pattern.compile(".*file not found.*", Pattern.CASE_INSENSITIVE);
+
+		Matcher m = regexFindFile.matcher(responce);
+
+		//need to clean the base case of the item we are looking for
+		String basecase = FilterResponce.removeItemCheckedFor(work.getBaseCaseObj().getBaseCase(), work.getItemToCheck());
+
+		if(m.find())
+		{
+			System.out.println("DEBUG Worker[" + threadId + "]: 404 for: " + url.toString());
+		}
+		else if(!responce.equalsIgnoreCase(basecase))
+		{
+		    sendResponseToManager(code, responce, rawResponce, basecase);
+		}
+	}
+
+	private void sendResponseToManager(int code, String responce, String rawResponce, String basecase) {
+		if(work.isDir())
+		{
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found Dir (base case)" + url.toString());
+		    }
+		    //we found a dir
+		    manager.foundDir(url, code, responce, basecase, rawResponce, work.getBaseCaseObj());
+		}
+		else
+		{
+		    //found a file
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found File (base case)" + url.toString());
+		    }
+		    manager.foundFile(url, code, responce, work.getBaseCaseObj().getBaseCase(), rawResponce, work.getBaseCaseObj());
+		}
+	}
+
+	private void parseHtml(HttpMethodBase httpMethod, String response) {
+		//parse the html of what we have found
+
+	    Header contentType = httpMethod.getResponseHeader("Content-Type");
+
+	    if(contentType != null)
+	    {
+	        if(contentType.getValue().startsWith("text"))
+	        {
+	            manager.addHTMLToParseQueue(new HTMLparseWorkUnit(response, work));
+	        }
+	    }
+		
+	}
+
+	private String getResponseAsString(HttpMethodBase httpMethod) throws IOException {
+		//set up the input stream 
+		BufferedReader input = new BufferedReader(new InputStreamReader(httpMethod.getResponseBodyAsStream()));
+		StringBuffer buf = new StringBuffer();
+		
+		String line;
+		
+		while((line = input.readLine()) != null) {		
+		    buf.append("\r\n" + line);
```

Comment:
Another easy win, the break line characters should be ```append```ed.
But there's a minor issue here, it appends ```\r\n``` even if the break line character might just be ```\n```, if not fixed, it should be noted with a FIXME task tag comment.